Libplanet/Blockchain/BlockChain.cs

```diff
@@ -472,6 +466,28 @@ internal BlockChain<T> Fork(HashDigest<SHA256> point)
                         break;
                     }
                 }
+
+                Block<T> pointBlock = Blocks[point];
+
+                var toUpdateAddresses = new HashSet<Address>();
+
+                for (
+                    Block<T> block = Tip;
+                    block.PreviousHash is HashDigest<SHA256> hash
```

Comment:
This condition won't work if `block` refers to a block right after genesis, which means addresses updated by the second block (where `blockIndex == 1`) are always excluded from the index.

Reply:
I think this condition is break when the block refers to the genesis block, not the second block. But I thought it is ok because every forking must have the genesis block.
I added tests for this case and found another bug and fixed it.